src/utp_stream.cpp

```diff
@@ -1084,25 +1084,43 @@ std::size_t utp_stream::read_some(bool const clear_buffers)
 	return ret;
 }
 
+std::size_t utp_stream::write_some(bool const clear_buffers)
```

Comment:
I think this patch (along with support for syncrhonous write) is also a good candidate for a separate PR that could land ahead of this one